server/project.go

```diff
@@ -404,6 +412,23 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 
 		repo := repoList[0]
 		repoPatch.ID = repo.ID
+
+		if repoPatch.FilePathTemplate != nil && repoPatch.BaseDirectory != nil {
```

Comment:
Use `||` instead of `&&`? Looks like we can only update one of them in `RepositoryPatch`.

And maybe we should merge the patch data with origin object firstly.